lua/mason-registry/init.lua

```diff
@@ -193,4 +193,23 @@ function M.refresh(cb)
     end
 end
 
+---table<string, string[]>
+local aliases = {}
+
+---Register aliases for the specified packages
+---@param new_aliases table<string, string[]>
+function M.register_package_aliases(new_aliases)
+    for key, value in pairs(new_aliases) do
+        aliases[key] = aliases[key] or {}
+        for _, alias in pairs(value) do
+            table.insert(aliases[key], alias)
+        end
+    end
+end
+
+---@param name string
+function M.get_package_aliases(name)
+    return aliases[name]
```

Comment:
I think it'd be nice for this function to always return a table:

```suggestion
    return aliases[name] or {}
```